File: 2023/cpp/day1/pt1.cpp
#include <iostream>
#include <string>
#include <vector>

void pt1(std::vector<std::string> dat) {
    int sum = 0;
    // for each line
    for (auto line : dat) {
        // std::cout << line << std::endl;
        // fwd pass
        int i, j;
        int a = 0;
        int b = 0;
        for (i = 0; i < line.length(); i++) {
            if (std::isdigit(line[i])) {
                auto digit = line[i];
                a = std::stoi(&digit);
                sum += 10 * a;
                break;
            }
        }

        // backwards pass
        for (j = line.length(); j >= i; j--) {
            if (std::isdigit(line[j])) {
                std::cout << line[j] << " " << b << std::endl;
                auto digit = line[j];
                b = std::stoi(&digit);
                sum += b;
                break;
            }
        }

        std::cout << line + "} " << line[i] << "-" << line[j] << std::endl;
        std::cout << 10 * a << "|" << b << ": " << 10 * a + b << std::endl;
    }
    printf("pt1: %d\n", sum);
}

File: 2023/rust/day1/src/pt1.rs
pub fn run(data: &String) {
    let radix = 10;
    // get nums in each line
    let mut line_nums = Vec::new();
    for line in data.lines() {
        let mut nums = Vec::new();
        for c in line.chars() {
            if c.is_numeric() {
                nums.push(c.to_digit(radix).unwrap())
            }
        }
        line_nums.push(nums);
    }

    // convert the nums in each line into numbers [1-99]
    let mut sum = 0;
    for nums in line_nums.iter() {
        let tens = 10 * nums[0];
        let ones = nums.last().unwrap();
        sum += tens + ones;
    }
    println!("pt1: {}", sum);
}
// fn pt1(data: &String) {
//     let radix = 10;
//     // get nums in each line
//     let mut line_nums = Vec::new();
//     for line in data.lines() {
//         let mut nums = Vec::new();
//         for c in line.chars() {
//             if c.is_numeric() {
//                 nums.push(c.to_digit(radix).unwrap());
//                 break;
//             }
//         }
//         for c in line.chars().rev() {
//             if c.is_numeric() {
//                 nums.push(c.to_digit(radix).unwrap());
//                 break;
//             }
//         }
//         line_nums.push(nums);
//     }
//
//     // convert the nums in each line into numbers [1-99]
//     let mut sum = 0;
//     for nums in line_nums.iter() {
//         let tens = 10 * nums[0];
//         let ones = nums.last().unwrap();
//         sum += tens + ones;
//     }
//     println!("pt1: {}\n", sum);
// }
